sdk/typescript/src/multisig/publickey.ts

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { fromB64, toB64 } from '@mysten/bcs';
+import { blake2b } from '@noble/hashes/blake2b';
+import { bytesToHex } from '@noble/hashes/utils';
+import { PublicKey, bytesEqual } from '../cryptography/publickey.js';
+import type {
+	SerializedSignature,
+	SignatureFlag,
+	SignatureScheme,
+} from '../cryptography/signature.js';
+import {
+	SIGNATURE_FLAG_TO_SCHEME,
+	SIGNATURE_SCHEME_TO_FLAG,
+	parseSerializedSignature,
+} from '../cryptography/signature.js';
+import { normalizeSuiAddress } from '../utils/sui-types.js';
+import { builder } from '../builder/bcs.js';
+// eslint-disable-next-line import/no-cycle
```

Comment:
this scares me a bit

Reply:
open to suggestions here, but this should be perfectly safe, most of the time cycles aren't an issue in libraries as long as the cyclic dependencies are used in functions that are not run as part of importing any of the modules involved in the cycle.

The alternative here would be to create a version of `publicKeyFromBytes` that doesn't know about multisigs or re-implement that logic as part of the multisig class, but the cycle as far as I know won't cause any issues. 

I added the eslint rule so we would flag these kinds of issues and make sure they made sense, but I don't think its always a bad thing